spec/services/statements/query_spec.rb

```diff
@@ -0,0 +1,265 @@
+RSpec.describe Statements::Query do
+  describe "#statements" do
+    let(:lead_provider) { create(:lead_provider) }
+
+    it "returns all statements" do
+      statement = create(:statement)
+      query = described_class.new
+
+      expect(query.statements).to eq([statement])
+    end
+
+    it "orders statements by payment date in ascending order" do
+      statement1 = create(:statement, payment_date: 2.days.ago)
+      statement2 = create(:statement, payment_date: 1.day.ago)
+      statement3 = create(:statement, payment_date: Time.zone.now)
+
+      query = described_class.new
+
+      expect(query.statements).to eq([statement1, statement2, statement3])
+    end
+
+    describe "filtering" do
+      describe "by lead provider" do
+        it "filters by lead provider" do
+          statement = create(:statement, lead_provider:)
+          _statement = create(:statement)
+          query = described_class.new(lead_provider:)
+
+          expect(query.statements).to eq([statement])
+        end
+
+        it "does not return statements for other Lead Providers" do
+          create(:statement)
+
+          query = described_class.new(lead_provider:)
+
+          expect(query.statements).to eq([])
+        end
+
+        it "does not filter by lead provider if none is supplied" do
+          condition_string = %(" WHERE \"lead_providers\".\"id")
+
+          query = described_class.new
+
+          expect(query.scope.to_sql).not_to include(condition_string)
```

Comment:
I would vote that we avoid these kind of assertions. It doesn't feel right to be asserting on raw SQL when we're two levels removed from it (by ActiveRecord and then the query service).

Reply:
agree although we did this before I'd rather we don't do this here as well

Reply:
💯